File: Miscellaneous/Pattern 1/Solution.java
import java.util.Scanner;

public class Pattern {
    public static void printPattern(int n) {
        int num = 1;
        for(int i = 1; i <= n; i++) {
            // Print leading spaces
            for(int space = 1; space <= n - i; space++) {
                System.out.print("   ");
            }
            
            // Print numbers
            if(i % 2 == 1) {  // Odd rows: left to right
                for(int j = 1; j <= n; j++) {
                    System.out.printf("%2d ", num++);
                }
            } else {  // Even rows: right to left
                int temp = num + n - 1;
                for(int j = 1; j <= n; j++) {
                    System.out.printf("%2d ", temp--);
                }
                num += n;
            }
            System.out.println();
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the value of n: ");
        int n = sc.nextInt();
        printPattern(n);
        sc.close();
    }
}

File: Miscellaneous/Pattern 1/Solution.py
def print_pattern(n):
    num = 1
    for i in range(1, n + 1):
        # Print leading spaces
        print("   " * (n - i), end="")
        
        # Print numbers
        if i % 2 == 1:  # Odd rows: left to right
            for j in range(n):
                print(f"{num:2d} ", end="")
                num += 1
        else:  # Even rows: right to left
            temp = num + n - 1
            for j in range(n):
                print(f"{temp:2d} ", end="")
                temp -= 1
            num += n
        print()

# Get input and call function
n = int(input("Enter the value of n: "))
print_pattern(n)